### PR TITLE
feat: support uniform bucket level access

### LINKED ISSUE #2869: Support "uniform bucket level access".
The `storage::BucketIamConfiguration` structure needs to change, eventually we will want to replace the existing `bucket_policy_only` field with `uniform_bucket_level_access`. But we do not want to drop compatibility, so consider this sequence:

- Rename `storage::BucketPolicyOnly` to `storage::UniformBucketLevelAccess`.

- Create a type alias `using storage::BucketPolicyOnly = storage::UniformBucketLevelAccess`.

- Add a new field to `storage::BucketIamConfiguration` called `uniform_bucket_level_access`.

- If the bucket metadata JSON object contains a `uniformBucketLevelAccess` object use that JSON object to populate both fields, and ignore any (now legacy) `bucketPolicyOnly` field.

- If the bucket metadata JSON object **only** contains the (now legacy) `bucketPolicyOnly` field, then use it to populate both fields.

This section is **unclear** discuss with @coryan and @frankyn before implementing it:

- When creating bucket metadata JSON objects for patch, insert, or update, use **only** the `uniform_bucket_level_access` field and only populate the `uniformBucketLevelAccess` field.

### PR DESCRIPTION
**Pull request by manish-qlogic**
This closes #2869

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/3186)

### CODE REVIEW

**Comment by coryan** [comment]
/cc: @JesseLovelace in case you want to take a look.

**Comment by manish-qlogic** [comment]
@JesseLovelace ,

Please let me know your comments on this PR.

**Comment by coryan** [comment]
@JesseLovelace any thoughts?

File: google/cloud/storage/bucket_metadata.h [hunk 1]
```diff
@@ -136,50 +136,51 @@ std::ostream& operator<<(std::ostream& os, CorsEntry const& rhs);
  * @warning this is a Beta feature of Google Cloud Storage, it is not subject
  *     to the deprecation policy and subject to change without notice.
  *
- * @see Before enabling Bucket Policy Only please review the
- *     [feature documentation][bpo-link], as well as
+ * @see Before enabling Uniform Bucket Level Access please
```
Comment (JesseLovelace) [inline google/cloud/storage/bucket_metadata.h]: Please update these to link to the UBLA documentation instead of the BPO documentation

Comment (coryan) [reply google/cloud/storage/bucket_metadata.h]: Can you give us a link?  Usable outside google?

Comment (coryan) [reply google/cloud/storage/bucket_metadata.h]: And for what is worth, there is already a bug (#3193) to fix that once the links are available.

**Pull request merged**